### PR TITLE
[ci skip]Add Skybox parser to js

### PR DESCRIPTION
**Pull request by pandamicro**
Original PR: https://github.com/cocos2d/cocos2d-x/pull/12804

### CODE REVIEW

**Comment by pandamicro** [comment]
@VisualSJ Please review

**Comment by pandamicro** [comment]
[ci rebuild]

File: cocos/scripting/js-bindings/script/studio/parsers/timelineParser-2.x.js [hunk 1]
```diff
@@ -1390,28 +1416,49 @@
...
+            var rightFileData = getSkyboxRes(json, "RightImage");
+            var upFileData = getSkyboxRes(json, "UpImage");
+            var downFileData = getSkyboxRes(json, "DownImage");
+            var forwardFileData = getSkyboxRes(json, "ForwardImage");
+            var backFileData = getSkyboxRes(json, "BackImage");
+
+            var skyBox = new jsb.Skybox();
+            skyBox.init(leftFileData,rightFileData,upFileData,downFileData,forwardFileData,backFileData);
+            skyBox.setCameraMask(cameraFlag);
+            node.addChild(skyBox, 0, "_innerSkyBox");
```
Comment (pandamicro) [inline cocos/scripting/js-bindings/script/studio/parsers/timelineParser-2.x.js:1461]: addChild(child, zindex, tag)
The tag should be a number, if you want to set the name of the child, it should be set with

`skyBox.setName("_innerSkyBox")`

**Comment by pandamicro** [comment]
@xiaofeng11 Review done, please check

File: cocos/scripting/js-bindings/script/studio/parsers/timelineParser-2.x.js (hunk 1 above)
Comment (pandamicro) [inline cocos/scripting/js-bindings/script/studio/parsers/timelineParser-2.x.js:1461]: Ok, I found that C++ addChild can pass string as the third parameter, we will refactor web engine for supporting it

**Comment by pandamicro** [comment]
Related task: https://github.com/cocos2d/cocos2d-x/issues/12944

File: cocos/scripting/js-bindings/script/studio/parsers/timelineParser-2.x.js (hunk 1 above)
Comment (xiaofeng11) [inline cocos/scripting/js-bindings/script/studio/parsers/timelineParser-2.x.js:1461]: Now C++ reader has changed to addChild(skyBox), I'll synchronize with C++ reader and create a new pr

Comment (pandamicro) [inline cocos/scripting/js-bindings/script/studio/parsers/timelineParser-2.x.js:1461]: I have merged this one, you can modify the logic based on the latest v3.7.1 branch, and please submit the PR to this branch too

Comment (xiaofeng11) [inline cocos/scripting/js-bindings/script/studio/parsers/timelineParser-2.x.js:1461]: OK, I'll submit new PR to this branch first, and I have just check v3.7.1 branch, seems skybox parser haven't been merged to it, should I submit a full skybox reader pr to v3.7.1?

Comment (xiaofeng11) [inline cocos/scripting/js-bindings/script/studio/parsers/timelineParser-2.x.js:1461]: Sorry, I have just make a mistake, I think this is 3.7 branch, the new parser will submit to v3.7.1

Comment (pandamicro) [inline cocos/scripting/js-bindings/script/studio/parsers/timelineParser-2.x.js:1461]: If you have any question, you can contact me via wechat

**Pull request merged**